### ISSUE TITLE
2016.11.5 IPv6 nameserver in resolv.conf leads to minion exception

### ISSUE THREAD

**Issue by RealKelsar** [saltstack/salt#41230]
### Description of Issue/Question
If resolv.conf contains a IPv6 nameserver entry, like "nameserver 2a00:1828:a000::dead", all states fail with an exception

### Setup
IPv6 in resolv.conf

### Steps to Reproduce Issue
On a Minion:
Put a IPv6 Nameserver in resolv.conf, salt-call state.sls $fuu

### Versions Report
Salt Version:
           Salt: 2016.11.5 (.4 is fine)
 
Dependency Versions:
           cffi: 1.10.0
       cherrypy: Not Installed
       dateutil: 2.6.0
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.9.6
        libgit2: 0.25.1
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 1.0.3
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: 1.2.5
      pycparser: 2.17
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: 0.25.0
         Python: 2.7.13 (default, Feb 21 2017, 12:21:12)
   python-gnupg: 0.3.8
         PyYAML: 3.12
          PyZMQ: 15.2.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: 0.2.4
        Tornado: 4.4.2
            ZMQ: 4.2.2
 
System Versions:
           dist: gentoo 2.3 
        machine: x86_64
        release: 4.9.27-gentoo
         system: Linux
        version: Gentoo Base System 2.3

### Result
```
[ERROR   ] An un-handled exception was caught by salt's global exception handler:
TypeError: can't serialize IPv6Address('fd00::ca0e:14ff:fea8:2344')
Traceback (most recent call last):
  File "/usr/lib/python-exec/python2.7/salt-call", line 11, in <module>
    salt_call()
  File "/usr/lib64/python2.7/site-packages/salt/scripts.py", line 379, in salt_call
    client.run()
  File "/usr/lib64/python2.7/site-packages/salt/cli/call.py", line 48, in run
    caller = salt.cli.caller.Caller.factory(self.config)
  File "/usr/lib64/python2.7/site-packages/salt/cli/caller.py", line 79, in factory
    return ZeroMQCaller(opts, **kwargs)
  File "/usr/lib64/python2.7/site-packages/salt/cli/caller.py", line 274, in __init__
    super(ZeroMQCaller, self).__init__(opts)
  File "/usr/lib64/python2.7/site-packages/salt/cli/caller.py", line 102, in __init__
    self.minion = salt.minion.SMinion(opts)
  File "/usr/lib64/python2.7/site-packages/salt/minion.py", line 669, in __init__
    self.gen_modules(initial_load=True)
  File "/usr/lib64/python2.7/site-packages/salt/minion.py", line 717, in gen_modules
    pillarenv=self.opts.get('pillarenv'),
  File "/usr/lib64/python2.7/site-packages/salt/pillar/__init__.py", line 162, in compile_pillar
    dictkey='pillar',
  File "/usr/lib64/python2.7/site-packages/salt/utils/async.py", line 75, in wrap
    ret = self._block_future(ret)
  File "/usr/lib64/python2.7/site-packages/salt/utils/async.py", line 85, in _block_future
    return future.result()
  File "/usr/lib64/python2.7/site-packages/tornado/concurrent.py", line 237, in result
    raise_exc_info(self._exc_info)
  File "/usr/lib64/python2.7/site-packages/tornado/gen.py", line 285, in wrapper
    yielded = next(result)
  File "/usr/lib64/python2.7/site-packages/salt/transport/zeromq.py", line 190, in crypted_transfer_decode_dictentry
    self._package_load(self.auth.crypticle.dumps(load)),
  File "/usr/lib64/python2.7/site-packages/salt/crypt.py", line 1267, in dumps
    return self.encrypt(self.PICKLE_PAD + self.serial.dumps(obj))
  File "/usr/lib64/python2.7/site-packages/salt/payload.py", line 181, in dumps
    return msgpack.dumps(msg, use_bin_type=use_bin_type)
  File "/usr/lib64/python2.7/site-packages/msgpack/__init__.py", line 47, in packb
    return Packer(**kwargs).pack(o)
  File "msgpack/_packer.pyx", line 231, in msgpack._packer.Packer.pack (msgpack/_packer.cpp:3661)
  File "msgpack/_packer.pyx", line 233, in msgpack._packer.Packer.pack (msgpack/_packer.cpp:3503)
  File "msgpack/_packer.pyx", line 192, in msgpack._packer.Packer._pack (msgpack/_packer.cpp:2657)
  File "msgpack/_packer.pyx", line 192, in msgpack._packer.Packer._pack (msgpack/_packer.cpp:2657)
  File "msgpack/_packer.pyx", line 192, in msgpack._packer.Packer._pack (msgpack/_packer.cpp:2657)
  File "msgpack/_packer.pyx", line 221, in msgpack._packer.Packer._pack (msgpack/_packer.cpp:3230)
  File "msgpack/_packer.pyx", line 228, in msgpack._packer.Packer._pack (msgpack/_packer.cpp:3382)
TypeError: can't serialize IPv6Address('fd00::ca0e:14ff:fea8:2344')
```

**Comment by gtmanfred** [comment]
It looks like we are fixing this here.

https://github.com/saltstack/salt/pull/41244

Thanks,
Daniel

**Comment by jcollie** [comment]
Did the patch for this make it into the final 2016.11.5 release?

**Comment by kyrias** [comment]
It was supposed to not be in the release since the tarballs had been made and uploaded to PyPI already, but should be patched in the repo.saltstack.com packages.

**Comment by jcollie** [comment]
After I posted the note I saw that the packages on repo.saltstack.com (at least the RHEL/CentOS 7 packages have it). The Fedora packages don't yet have the patch though.

**Comment by rvstaveren** [comment]
In my situation, it prevents minion startup altogether with the following message:

```
2017-06-02 11:58:02,725 [salt.minion                                                ][ERROR   ][84749] Error while bringing up minion for multi-master. Is master at salt responding?
```

I'm letting the minion run with all defaults:

```
# cat /usr/local/etc/salt/minion | sed -e 's/#.*$//' | grep -v '^$'
include: /usr/local/etc/salt/grain
```

Minion is as follows:

```
[root@insights /usr/home/ruben]# salt-call --local test.versions
[WARNING ] Although 'dmidecode' was found in path, the current user cannot execute it. Grains output might not be accurate.
local:
    Salt Version:
               Salt: 2016.11.5
     
    Dependency Versions:
               cffi: Not Installed
           cherrypy: Not Installed
           dateutil: 2.6.0
          docker-py: Not Installed
              gitdb: Not Installed
          gitpython: Not Installed
              ioflo: Not Installed
             Jinja2: 2.9.5
            libgit2: Not Installed
            libnacl: Not Installed
           M2Crypto: Not Installed
               Mako: Not Installed
       msgpack-pure: Not Installed
     msgpack-python: 0.4.7
       mysql-python: Not Installed
          pycparser: Not Installed
           pycrypto: 2.6.1
       pycryptodome: Not Installed
             pygit2: Not Installed
             Python: 2.7.13 (default, Apr 29 2017, 01:14:04)
       python-gnupg: Not Installed
             PyYAML: 3.11
              PyZMQ: 16.0.2
               RAET: Not Installed
              smmap: Not Installed
            timelib: Not Installed
            Tornado: 4.5.1
                ZMQ: 4.1.5
     
    System Versions:
               dist:   
            machine: amd64
            release: 10.3-RELEASE-p18
             system: FreeBSD
            version: Not Installed
```

**Comment by gtmanfred** [comment]
Are you installing with pip or with the package?

We have asked the community maintainers to add this patch to revert those
changes.

https://github.com/saltstack/salt/pull/41244

So if they have done that, this should work, if not you will need to apply
it.

We have this fixed for 2016.11.6

On Fri, Jun 2, 2017 at 6:00 AM, Ruben van Staveren <notifications@github.com

**Comment by RealKelsar** [comment]
Seems fixed to me.
